File: TempAddWait.java
import java.nio.file.*;
import java.util.*;

public class TempAddWait {
    public static void main(String[] args) throws Exception {
        Path path = Paths.get("c:\\Users\\User\\IdeaProjects\\mobileauto\\src\\main\\java\\org\\example\\Loyalty.java");
        List<String> lines = Files.readAllLines(path);
        List<String> outLines = new ArrayList<>();

        for (int i = 0; i < lines.size(); i++) {
            String line = lines.get(i);
            String stripped = line.trim();
            
            if (stripped.startsWith("System.out.println") && 
                (stripped.contains("Клик") || stripped.contains("Ввод") || stripped.contains("Проверка"))) {
                
                boolean hasSleep = false;
                int j = i - 1;
                while (j >= 0) {
                    String sj = lines.get(j).trim();
                    if (sj.isEmpty() || sj.startsWith("//")) {
                        j--;
                        continue;
                    }
                    if (sj.startsWith("System.out.println")) {
                        // ignore print statements when searching upwards for a sleep
                        // so that chains of prints don't hide the missing sleep
                        j--;
                        continue;
                    }
                    if (sj.contains("Thread.sleep")) {
                        hasSleep = true;
                    }
                    break;
                }
                
                if (!hasSleep) {
                    // get indent
                    int indentLen = line.length() - line.stripLeading().length();
                    String indent = line.substring(0, indentLen);
                    outLines.add(indent + "Thread.sleep(1000);");
                }
            }
            outLines.add(line);
        }
        
        Files.write(path, outLines);
        System.out.println("Modified Loyalty.java successfully.");
    }
}

File: temp_add_wait.py
import sys

filepath = r"c:\Users\User\IdeaProjects\mobileauto\src\main\java\org\example\Loyalty.java"

with open(filepath, "r", encoding="utf-8") as f:
    lines = f.readlines()

out_lines = []
for i, line in enumerate(lines):
    stripped = line.strip()
    if stripped.startswith("System.out.println") and any(k in stripped for k in ["Клик", "Ввод", "Проверка"]):
        has_sleep = False
        j = i - 1
        while j >= 0:
            s_j = lines[j].strip()
            if not s_j:
                j -= 1
                continue
            if s_j.startswith("//"):
                j -= 1
                continue
            if s_j.startswith("System.out.println"):
                # Ignore other prints just above it (like "✅ ...") when searching for sleeps
                j -= 1
                continue
            if "Thread.sleep" in s_j:
                has_sleep = True
            break
        
        if not has_sleep:
            indent = line[:len(line) - len(line.lstrip())]
            out_lines.append(f"{indent}Thread.sleep(1000);\n")
            
    out_lines.append(line)

with open(filepath, "w", encoding="utf-8") as f:
    f.writelines(out_lines)

print("Modification complete.")
